tools/RAiDER/models/weatherModel.py

```diff
@@ -212,6 +218,26 @@ def load(
             self._getZTD(zref)
             return None
 
+
+    def _convert_z_to_ellipsoid(self): 
+        '''Convert original geoidal heights to ellipsoidal'''
+        bounds = (self._ll_bounds[2], self._ll_bounds[0], self._ll_bounds[3], self._ll_bounds[1])
+        geoid_arr, geoid_profile = read_geoid('egm_96',extent=bounds,force_agi_read=True)
+        shift = -0.5
+        geoid_profile = translate_profile(geoid_profile, shift, shift)
+        geoid_transformer = Transformer.from_crs(geoid_profile['crs'], self._proj, always_xy = True) 
+        gt = tuple(geoid_profile['transform'])
+        Nx = geoid_profile['width']
+        Ny = geoid_profile['height']
+        geoid_x = [gt[2] + gt[0]*k for k in range(Nx)]
+        geoid_y = [gt[5] + gt[4]*k for k in range(Ny)]
+        geoid_y.reverse()
+        new_x, new_y, new_arr = geoid_transformer.transform(geoid_x, geoid_y, np.flipud(geoid_arr))
+        F = rgi((new_y, new_x), new_arr, method='linear', bounds_error=False)
+        ellip_correction = F((self._ys, self._xs))
+        self._zs = self._zs + ellip_correction
+
+
```

Comment:
This seems to be the place where I can comment. Overall looks great.

So, I assume that the geo-metadata was not possible to retrieve from the original weather model files [1]. That said, my only suggestion is to distill the method that is being used for reprojection and encapsulate that in a function. For example, `reproject(src_arr, src_transform, dst_transform)`. Would be easier to test and debug down the line. I am not familiar with the array operations and assume they are code you have used elsewhere. My experience with pyproj is that it is meant for vector data with `(x, y, z)` - so interesting to see this. Not sure of the speed differences with gdal/rasterio.

Looks like the broadcasting is correct and so that's great.

[1] https://rasterio.readthedocs.io/en/latest/topics/datasets.html (e.g.`ds = rasterio('NETCDF:<path_to_file>:/path/to/group')`.

Reply:
The other parts of the code look to be cleanup and I trust that you are doing this to improve the overall performance and may have been integrated from other pending pull requests.